tests/Resources/ti.require.test.js

```diff
@@ -0,0 +1,277 @@
+	it.windowsDesktopBroken('should load a node module by id in node_modules folder living at same level as requirer', function () {
+		var abbrev = require('abbrev');
+		should(abbrev).be.a.Function;
+		should(abbrev('foo', 'fool', 'folding', 'flop')).eql({ fl: 'flop', flo: 'flop', flop: 'flop', fol: 'folding', fold: 'folding', foldi: 'folding', foldin: 'folding', folding: 'folding', foo: 'foo', fool: 'fool' });
+	});
+
+	// FIXME We have no native ti.touchid module for windows!
+	it.windowsMissing('loads native module by id', function () {
+		var object = require('ti.touchid');
+		should(object).have.property('apiName');
+		// Of course, the module's apiName is wrong, so we can't test that
+		// should(object.apiName).be.eql('ti.touchid');
+		should(object).have.property('authenticate');
+	});
+
+	// TODO Add a test for requiring a node module up one level from requiring file!
+
+	it.windowsPhoneBroken('loads path using legacy fallback if first segment matches native module id and wasn\'t found inside module', function () {
+		var object = require('ti.touchid/example');
+		should(object).have.property('name');
+		should(object.name).be.eql('ti.touchid/example.js');
+	});
+
+	it('require from node_modules should not break the app', function () {
+		var object = require('bar'),
+			baz;
+		should(object).have.property('name');
+		should(object.name).be.eql('bar');
+		should(object).have.property('baz');
+		baz = object.baz;
+		should(baz).have.property('foo');
+		should(baz.foo.filename).be.eql('/node_modules/foo/index.js');
+	});
+
+	it('require should prefer closest node_modules', function () {
+		// require('bar') will require package baz under node_modules/bar/node_modules
+		// require('bax') should not point to this version of baz but to node_modules/baz
+		var object = require('bar'),
+			baz,
+			baz2;
+		should(object).have.property('name');
+		should(object.name).be.eql('bar');
+		should(object).have.property('baz');
+		baz = object.baz;
+		should(baz.name).be.eql('baz');
+		should(baz.filename).be.eql('/node_modules/bar/node_modules/baz/index.js');
+		should(baz.dirname).be.eql('/node_modules/bar/node_modules/baz');
+		baz2 = require('baz');
+		should(baz2.name).be.eql('baz');
+		should(baz2.filename).be.eql('/node_modules/baz/index.js');
+		should(baz2.dirname).be.eql('/node_modules/baz');
+	});
+
+	it('require should not fail when exports is falsey', function () {
+		var files = [
+			{ filename: 'empty-double', expected: 0 },
+			{ filename: 'empty-single', expected: 0 },
+			{ filename: 'false', expected: 0 },
+			{ filename: 'nan', expected: 0 },
+			{ filename: 'null', expected: 0 },
+			{ filename: 'undefined', expected: 0 },
+			{ filename: '0', expected: 0 }
+		];
+		for (var obj in files) {
+			var result;
+			should.doesNotThrow(
+				// eslint-disable-next-line no-loop-func
+				function () {
+					result = require('./fixtures/' + obj.filename);
+				}
+			);
+			should(result).be.eql(obj.expected);
```

Comment:
why are we always expecting 0 as the result? I assume because of the `eql` method being equivalent to `==`?

I think we should likely use `should(result).be.exactly(obj.expected);` and compare to '', false, null, undefined, 0;

For NaN, we can't compare vis `==` or `===`, but should use `isNaN(result).should.be.true();`